src/kaleidoscope/HardwareDescription.h

```diff
@@ -0,0 +1,33 @@
+/* -*- mode: c++ -*-
+ * HardwareDescription -- Hardware description base class
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "kaleidoscope/driver/scanner/NoOpScanner.h"
+#include "kaleidoscope/driver/led/NoLeds.h"
+#include "kaleidoscope/driver/mcu/NoMCU.h"
+
+namespace kaleidoscope {
+
+struct HardwareDescription {
+  typedef kaleidoscope::driver::scanner::NoOpScannerDescription ScannerDescription;
+  typedef kaleidoscope::driver::scanner::NoOpScanner<ScannerDescription> KeyScanner;
+  typedef kaleidoscope::driver::led::NoLeds LEDs;
+  typedef kaleidoscope::driver::mcu::NoMCU MCU;
```

Comment:
> Shouldn't we also have a `LEDsDescription` and `MCUDescription` here to make things more versatile for future usage?

We probably should, yes.

> Which LEDs are meant by the component LEDs (per-key LEDs, other LEDs - like on the ErgoDox EZ - or underglow)? Are these all meant to be covered by the LEDs module?

This is also something we should figure out :)

At one point, I was thinking about having a separate per-key LED controller, another for underglow, side-glow, etc. But then we'd have a ton of separate controllers that share a ton of code and APIs, but appear distinct for the programmer. So my current thinking is that this will all go under a single LED driver, and how that further separates things, will be up to the hardware plugin. This allows us the greatest flexibility. I have a few ideas about this, but that'll be a separate topic.

> `...::led::NoLEDs` sounds more consistent to me.

 We have a few things named `LEDs`, and other things as `Leds`. I'd favour the former, but my impression is that the second is more common in the Arduino world. @obra, do you have any suggestions here?

Reply:
Yeah, to me nesting controllers seems the right way. Having a `LEDs` controller that contains other sub-controllers that are dedicated to different types of LEDs.

Reply:
I'll resolve this conversation, and defer figuring out LED-related naming to a future PR/issue. Well, kinda... we still want a name for what the new API introduces, and `LedDriver` seems to be winning now. But any conceptual change, like deciding whether this driver controls all leds, or just per-key leds, etc - that's something we'll have to decide later.